### DISCUSSION TITLE
增加文字大字滚屏显示功能

### DISCUSSION THREAD

**Discussion by OrangeFish6** [ClassIsland/ClassIsland#562] [插件功能请求]
### 检查清单

- [X] 我已将本应用更新到最新测试版 ![最新的版本号](https://img.shields.io/github/v/release/ClassIsland/ClassIsland?include_prereleases&style=flat-square&label=)，并看过[最新提交](https://github.com/ClassIsland/ClassIsland/commits/dev/)，确认我想要的功能还没有实现。
- [X] 我已在 [Issues](https://github.com/ClassIsland/ClassIsland/issues?q=label%3A新功能) 和 [ClassIsland 文档](https://docs.classisland.tech/app/)中检索，确认我想要的功能未被提交过。
- [X] 我认为我想要的功能是用户广泛需要的，插件不能替代的，且没有超出 ClassIsland 作为**灵动课表**的开发目标，而非添加与课表及教学辅助无关的内容。（您可以学习[编写插件](https://docs.classisland.tech/dev/plugins/create-project/)来实现特殊的需求）。
- [X] 我已知晓由于开发者时间精力有限，加上已经积压了大量的功能请求，我提交的功能请求可能需要比较久的时间才能实现。

### 背景与动机

班里常常会有些公告，需要一段时间的显示。有些时候黑板写满重要内容，无法擦除，口头通报又会造成部分同学因听不清造成的误解。

### 想要实现或优化的功能

文字大字显示功能（类似于目前的天气预警提示功能的效果）
在状态栏打开的快捷栏中可以增设一个公告板的入口，里面可以自行设置以下两种效果：
1.文字固定显示：输入文字，可自行设置停留时间，可自行设置是否会被上课提醒/下课提醒/天气预警等打断
2.文字滚动显示：输入文字，可自行设置滚动速度、滚动次数，可自行设置是否会被上课提醒/下课提醒/天气预警等打断

### 最后一步

- [X] 我认为上述的描述已经足以详细，以便开发人员理解您想要的功能。如果我的 issue 没有按照上述的要求填写，可能会被无条件关闭。

**Comment by DryIce-cc** [comment]
这个功能有点超出 ClassIsland 的开发目标，您可以通过编写插件实现。
也可以使用 ZongziTek ~或 MooncakeTek~ 的小黑板代替。
您提到的提醒「公告板」效果，计划将会在「行动」功能中提供接口，届时开发此插件应该会简单一点？

**Comment by LiPolymer** [comment]
您提到的内容将在ExtraIsland的弹出式通知内实现 不过时间安排困难 开发周期较长 还请见谅

**Comment by ghost** [comment]
现在的ci1.7版本其实有可供使用的滚动提醒api，也许应该在行动中加上？

**Reply by DryIce-cc** [reply]
我看看

**Reply by ghost** [reply]
可以看看我的仓库 Easy-Notification/EasyNotification

https://github.com/Easy-Notification/EasyNotification/blob/master/EasyNotification%2FServices%2FNotificationProviders%2FEasyNotificationProvider.cs#L125-L143

~~抄的集控提醒提供方的代码~~
